specs/lightclient/sync-protocol.md

```diff
@@ -78,10 +80,10 @@ class LightClientUpdate(Container):
     header: BeaconBlockHeader
     # Next sync committee corresponding to the header
     next_sync_committee: SyncCommittee
-    next_sync_committee_branch: Vector[Bytes32, log2(NEXT_SYNC_COMMITTEE_INDEX)]
+    next_sync_committee_branch: Vector[Bytes32, NEXT_SYNC_COMMITTEE_INDEX.bit_length()]
```

Comment:
Replacing log2 with bit_length would actually give an off-by-one error. Why not just define a function `log2`? We would need it in a lot of contexts anyway.

Reply:
@vbuterin 

1. You're right, `get_subtree_index` is off. Should be `uint64(generalized_index % 2**((generalized_index).bit_length() - 1))`. Thanks!
2. By `log2`, do you mean `ceillog2` (existing) or define a new `floorlog2`?
    - We used `ceillog2` to compute the depth from the chunkified items count.
    - But here we are computing the depth from `generalized_index` so isn't `generalized_index.bit_length()` result exactly the depth of the given leaf? (No mix-in here, right?)

Reply:
🙏
Fixed in https://github.com/ethereum/eth2.0-specs/pull/2147/commits/d01a4ad82374dbfbabf047315536f90fe90bf928 with sanity tests.